#! /bin/sh /usr/share/dpatch/dpatch-run
## check-ast-runnning.dpatch by Diego Iastrubni <diego.iastrubni@xorcom.com>
##
## DP: Display a warning when asterisk is not responding to the manager.
## DP: This usually means asterisk is down

@DPATCH@
diff -urNad freepbx-2.1.1/amp_conf/htdocs/admin/config.php /tmp/dpep.NjMnkU/freepbx-2.1.1/amp_conf/htdocs/admin/config.php
--- freepbx-2.1.1/amp_conf/htdocs/admin/config.php	2006-05-12 02:40:18.000000000 +0300
+++ /tmp/dpep.NjMnkU/freepbx-2.1.1/amp_conf/htdocs/admin/config.php	2006-07-09 14:12:58.000000000 +0300
@@ -189,6 +189,28 @@
 	case 'modules':
 		include 'page.modules.php';
 	break;
+
+	case '':
+		// on the main page, alert the user if asterisk is not running
+		// try to reuse as much strings as needed
+		require_once('common/php-asmanager.php');
+		$astman = new AGI_AsteriskManager();
+		if ($res = $astman->connect("127.0.0.1", $amp_conf["AMPMGRUSER"] , $amp_conf["AMPMGRPASS"])) {
+			$astman->disconnect();
+		}
+		else{
+			echo "<style>.clsError{ border: #BB0A0A 1px solid; background-color: #ffc0c0; }</style>\n";
+			echo "<p><div class='clsError'>\n";
+			echo "<b>" . _("Warning:") . "</b>\n";
+			echo "<br>";
+			echo "<br>\n";
+			echo _("Cannot connect to Asterisk Manager with ").$amp_conf["AMPMGRUSER"];
+			echo "<br>";
+			echo _("Asterisk may not be running.");
+			echo "</div></p>\n";
+		}
+	break;
+
 }
 
 //use main translation file for footer
